libcontainer/nsenter/nsexec.c

```diff
@@ -841,6 +851,12 @@ void nsexec(void)
 					bail("setgroups failed");
 			}
 
+			/* Unshare CLONE_NEWIPC after becoming root in user namespace */
+			if ((config.cloneflags & CLONE_NEWUSER) && (config.cloneflags & CLONE_NEWIPC)) {
+				if (unshare(CLONE_NEWIPC) < 0)
+					bail("unshare ipc failed");
+			}
```

Comment:
I think that the reasoning is that you need to have this run after `setuid(0)` and `setgid(0)`. #975 was meant to make it possible to have this section earlier.

@mrunalp @dqminh Do you mind if I carry this and #975 and make a new PR that combines both?